src/components/common/UserProfileEdit/UserAdvanceSettings.tsx

```diff
@@ -0,0 +1,243 @@
+    id: 'users.UserProfileEdit.UserAdvanceSettings.heading',
+    defaultMessage: 'Advanced settings {learnMoreLink}',
+  },
+  metaDescription: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.metaDescription',
+    defaultMessage: `To connect directly to Gnosis chain and pay for your own transactions, disable this option.`,
+  },
+  metaDescGlobalOff: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.metaDescGlobalOff',
+    defaultMessage: `Metatransactions are disabled globally.`,
+  },
+  labelMetaTx: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.labelMetaTx',
+    defaultMessage: `Metatransactions ({isOn, select,
+      true {active}
+      other {inactive}
+    })`,
+  },
+  metaTooltip: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.metaTooltip',
+    defaultMessage: `Metatransactions are turned on by default.
+    If you would rather connect directly to the chain,
+    and pay for your own transactions, you can turn them off
+    by switching the toggle at any time. {br}{br} Please note,
+    this setting is stored locally in your browser,
+    if you clear your cache you will need to turn Metatransactions off again.`,
+  },
+  customEndpoints: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.customEndpoints',
+    defaultMessage: `Enable custom endpoints ({isOn, select,
+      true {active}
+      other {inactive}
+    })`,
+  },
+  endpointsDescription: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.metaDescription',
+    defaultMessage: `If you prefer maximum decentralisation, you may use your own custom endpoints for Colony.`,
+  },
+  labelGnosisRPC: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.labelGnosisRPC',
+    defaultMessage: 'Gnosis Chain RPC',
+  },
+  gnosisRPCTooltip: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.gnosisRPCTooltip',
+    defaultMessage: `You will be able to toggle the Gnosis Chain RPC on once you have successfully validated that the endpoint works.`,
+  },
+  validate: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.validate',
+    defaultMessage: 'Validate',
+  },
+  snackbarSuccess: {
+    id: 'users.UserProfileEdit.UserAdvanceSettings.snackbarSuccess',
+    defaultMessage: 'Profile settings have been updated.',
+  },
+});
+
+interface FormValues {
+  [SlotKey.Metatransactions]: boolean;
+  [SlotKey.DecentralizedMode]: boolean;
+  [SlotKey.CustomRPC]: string;
+}
+
+const validationSchema = yup.object({
+  [SlotKey.Metatransactions]: yup.bool(),
+  [SlotKey.DecentralizedMode]: yup.bool(),
+  [SlotKey.CustomRPC]: yup.string().url(),
+});
+
+const displayName = 'users.UserProfileEdit.UserAdvanceSettings';
+
+const UserAdvanceSettings = () => {
+  const [showSnackbar, setShowSnackbar] = useState<boolean>(false);
+  useEffect(() => {
+    if (showSnackbar) {
+      const timeout = setTimeout(() => setShowSnackbar(true), 200000);
+      return () => {
+        clearTimeout(timeout);
+      };
+    }
+    return undefined;
+  }, [showSnackbar]);
```

Comment:
I'm not sure about the purpose of this `useEffect`. The timeout for hiding the Snackbar seems to already be present inside that component. What is this doing?

Reply:
Your are correct, that was a leftover from the dapp